snapcraft/storeapi/v2/channel_map.py

```diff
@@ -0,0 +1,266 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2020 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from typing import Any, Dict, List, Optional, Set, Union
+
+"""
+This module holds representations for results for the v2 channel-map
+API endpoint provided by the Snap Store.
+
+The full API is documented on
+https://dashboard.snapcraft.io/docs/v2/en/snaps.html#snap-channel-map
+"""
+
+
+class Progressive:
+    """
+    Represents Progressive information for a MappedChannel.
+    """
+
+    @classmethod
+    def unmarshal(cls, payload: Dict[str, Union[str, bool]]) -> "Progressive":
+        return cls(
+            key=payload["key"],
+            paused=payload["paused"],
+            percentage=payload["percentage"],
+        )
+
+    def marshal(self) -> Dict[str, Union[str, bool]]:
+        return {"key": self.key, "paused": self.paused, "percentage": self.percentage}
+
+    def __repr__(self) -> str:
+        return f"<Progressive: {self.percentage!r}>"
+
+    def __init__(
```

Comment:
should __init__ be the first instance method?

Reply:
we never discussed this, this is the order I usually see in the wild, I may be wrong or biased. What is your preference?

Reply:
Either way, I just thought of classmethods before `__init__` guideline, and just assumed that implied that __init__ should be top.

Reply:
We should probably rename it to special attributes. Thoughts?

Reply:
Googling about, I found some interesting rationale wrt ordering:
https://stackoverflow.com/a/36967648